File: examples/alg_surface.py
#
# jython examples for jas.
# $Id$
#

import sys

from java.lang import System
from java.lang import Integer

from jas import Ring
from jas import PolyRing
from jas import Ideal
from jas import QQ, AN, RF, EF
from jas import terminate
from jas import startLog

# polynomial examples: 

Yr = EF(QQ()).extend("x,y").extend("z","z^2 + x^2 + y^2 - 1").build();
#print "Yr    = " + str(Yr);
#print

[one,x,y,z] = Yr.gens();
print "one   = " + str(one);
print "x     = " + str(x);
print "y     = " + str(y);
print "z     = " + str(z);
print;

f = (1+z)*(1-z); # / ( x**2 + y**2 );
print "f     = " + str(f);
print;

g = f / (1 - z); 
print "g     = " + str(g);
print;

#startLog();
terminate();

File: examples/alg_surface.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# polynomial examples: 

Yr = EF.new(QQ()).extend("x,y").extend("z","z^2 + x^2 + y^2 - 1").build();
#puts "Yr    = " + str(Yr);
#puts

one,x,y,z = Yr.gens();
puts "one   = " + str(one);
puts "x     = " + str(x);
puts "y     = " + str(y);
puts "z     = " + str(z);
puts;

f = (1+z)*(1-z); # / ( x**2 + y**2 );
puts "f     = " + str(f);
puts;

g = f / (1 - z); 
puts "g     = " + str(g);
puts;

#startLog();
terminate();
